src/coreclr/nativeaot/Runtime/ThunksMapping.cpp

```diff
@@ -121,8 +121,14 @@ EXTERN_C NATIVEAOT_API void* __cdecl RhAllocateThunksMapping()
         return NULL;
     }
 
-#if defined(HOST_OSX) && defined(HOST_ARM64)
+#if defined(HOST_ARM64)
+#if defined(HOST_MACCATALYST) || defined(HOST_IOS) || defined(HOST_TVOS)
+    RhFailFast(); // TODO: add pthread_jit_write_protect_np workaround
+#elif defined(HOST_OSX)
     pthread_jit_write_protect_np(0);
+#else
+    #error "Unknown OS"
```

Comment:
Wouldn't this throw on any other host OS?

Reply:
ah right, I missed this while refactoring, thanks!